src/socks.rs

```diff
@@ -7,7 +7,24 @@ use tokio::{
 
 use crate::tcp::{self, NetStream};
 
-pub async fn handle_connection(stream: NetStream) -> Result<()> {
+#[derive(Clone)]
+pub struct AuthInfo {
+    pub user: String,
+    pub pass: String,
+}
+
+impl AuthInfo {
+    pub fn new(s: String) -> Self {
+        let (user, pass) = s.split_once(':').unwrap();
```

Comment:
`user` and `pass` can be generated as random strings if `s` equals to `rand` or other string